cmd/open.go

```diff
@@ -17,15 +27,95 @@ Pass the path to the directory that contains the solution you want to see on the
 	`,
 	Args: cobra.ExactArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
+		cfg := config.NewConfig()
+
+		v := viper.New()
+		v.AddConfigPath(cfg.Dir)
+		v.SetConfigName("user")
+		v.SetConfigType("json")
+		// Ignore error. If the file doesn't exist, that is fine.
+		_ = v.ReadInConfig()
+		cfg.UserViperConfig = v
+
+		return runOpen(cfg, cmd.Flags(), args)
+	},
+}
+
+func runOpen(cfg config.Config, flags *pflag.FlagSet, args []string) error {
+	var url string
+
+	if remote, _ := flags.GetBool("remote"); remote {
+		usrCfg := cfg.UserViperConfig
+
+		apiUrl := fmt.Sprintf("%s/solutions/latest", usrCfg.GetString("apibaseurl"))
+
+		client, err := api.NewClient(usrCfg.GetString("token"), usrCfg.GetString("apibaseurl"))
+		if err != nil {
+			return err
+		}
+
+		req, err := client.NewRequest("GET", apiUrl, nil)
+		if err != nil {
+			return err
+		}
+
+		q := req.URL.Query()
+		q.Add("exercise_id", args[0])
+		if track, _ := flags.GetString("track"); track != "" {
+			q.Add("track_id", track)
+		}
+		req.URL.RawQuery = q.Encode()
+
+		res, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+
+		defer res.Body.Close()
+		var payload openPayload
+		if err := json.NewDecoder(res.Body).Decode(&payload); err != nil {
+			return fmt.Errorf("unable to parse API response - %s", err)
+		}
+
+		if res.StatusCode != http.StatusOK {
+			switch payload.Error.Type {
+			case "track_ambiguous":
+				return fmt.Errorf("%s: %s", payload.Error.Message, strings.Join(payload.Error.PossibleTrackIDs, ", "))
```

Comment:
#460 suggests a prompt such as this:
```
We found more than one. Which one did you mean?
Type the number of the one you want to select it.

  [1] python/clock
  [2] go/clock
  [3] ruby/clock
  [4] elixir/clock
  [5] haskell/clock

> 
```

Rather than following this approach, I decided to stay consistent with the way `download` handles the `track_ambiguous` error and that is just by returning the error.

Reply:
I agree. I need to go through the old issues and get rid of all the references to the interactive stuff. It's too complicated and makes the whole thing harder to use.